src/faction_camp.cpp

```diff
@@ -2385,43 +2375,40 @@ void talk_function::camp_hunting_results( int skill, const std::string &task, in
 }
 
 // map manipulation functions
-bool talk_function::om_camp_upgrade( npc &comp, const tripoint &omt_pos )
+bool basecamp::om_upgrade( npc &comp, const std::string &next_upgrade, const tripoint &upos )
 {
     editmap edit;
 
-    oter_id &omt_ref = overmap_buffer.ter( omt_pos );
-    std::string om_old = omt_ref.id().c_str();
-
-    if( !edit.mapgen_set( om_next_upgrade( om_old ), tripoint( omt_pos ) ) ) {
+    if( !edit.mapgen_set( next_upgrade, upos ) ) {
         return false;
     }
 
-    tripoint np = tripoint( 5 * SEEX + 5 % SEEX, 4 * SEEY + 5 % SEEY, comp.posz() );
-    np.x = g->u.posx() + 1;
-    np.y = g->u.posy();
+    if( upos == pos ) {
+        tripoint np = tripoint( 5 * SEEX + 5 % SEEX, 4 * SEEY + 5 % SEEY, pos.z );
+        np.x = g->u.posx() + 1;
```

Comment:
This looks strange. The line above initializes `np.x` and `np.y` with specific values, but here you overwrite them again. Why not put those values in the constructor call above?

Reply:
Really just copying acidia's code again here.
Fixed.